module/VuFind/src/VuFind/Sitemap/Sitemap.php

```diff
@@ -65,11 +104,31 @@ public function __construct($frequency = 'weekly')
      */
     protected function getEntry($url)
     {
-        $loc = htmlspecialchars($url);
+        $alternativeLinks = '';
+        if ($this->alternativeLanguages) {
+            $lngParam = strpos($url, '?') === false ? '?lng=' : '&lng=';
```

Comment:
If we have a sitemap plugin that provides URLs for code that lives outside of VuFind, do we need to worry about this logic inappropriately manipulating URLs? I haven't traced all the relationships thoroughly to understand exactly how things might behave... but this seems like some very VuFind-specific behavior being added to a class that was meant to be a fairly abstract representation of an XML sitemap. Would it be better to implement this with a callback function for applying language codes to URLs, or something like that?

Reply:
Added option for a plugin to disable language versions. Plugins can now also define frequency that overrides the default one. And they can now return per-url languages and/or frequencies for maximum flexibility.